### DISCUSSION TITLE
No telegrams received, RTL-SDR V4 Dongle, Installation via make (complete description of way for verifying in text)

### DISCUSSION THREAD

**Discussion by Milvus11** [wmbusmeters/wmbusmeters#1228] [Q&A]
Hey guys,
I am a real newbie, I've been searching through the internet, also used AI for help, but don't receive any telegrams for my Techem FM radio 4 water meters, nor my vario 4 Typ 4.1.2 floor heating meter, though trying it for two weeks now. I had some trouble installing it correctly so I made myself a manual for installation via make, to verify it was right and nothing went wrong there already, I attach it here as well. Underneath you will find the status and log output. Can someone tell me if/what I did wrong?

### 1. System update
`sudo apt update`
`sudo apt upgrade -y`

### 2. Install dependencies
`sudo apt install -y git cmake build-essential librtlsdr-dev libusb-1.0-0-dev`

### 3. Install RTL-SDR
Blocking intern DBT-T
`echo "blacklist dvb_usb_rtl28xxu" | sudo tee /etc/modprobe.d/blacklist-rtl.conf `
Reboot Raspi
`sudo reboot`
Install rtl-sdr
`sudo apt install -y rtl-sdr`
Test, if RTL-SDR dongle is recognised
`rtl_test`

Stopping with "CTRL + C"

### 4. Install wmbusmeters

install libxm12-dev
`sudo apt install -y libxml2-dev`

clone wmbusmeters repository
`git clonehttps://github.com/wmbusmeters/wmbusmeters.git `

change to repository
`cd wmbusmeters `

make programm
`make`

install wmbusmeters
`git clone https://github.com/xaelsouth/rtl-wmbus.git`
`cd rtl-wmbus`
`make`
`sudo make install`

configure wmbusmeters.conf file
`sudo nano /etc/wmbusmeters.conf `

Insert following configuration:
```
loglevel=normal
# Remember to change auto here to the device you are going to use in production.
device=rtlwmbus:868.95M
Listento=c1, t1
#device=auto:t1
logfile=/var/log/wmbusmeters/wmbusmeters.log
# Set to true to capture all received telegrams in log file.
logtelegrams=true
format=json
```

start wmbusmeters service
`sudo systemctl start wmbusmeters `

to check the status:
`sudo systemctl status wmbusmeters `

check the reveived data:
`tail -f /var/log/wmbusmeters/wmbusmeters.log`

-----

However, I don't receive any data. The satus check gives me following output:

```wmbusmeters.service - "wmbusmeters service"
     Loaded: loaded (/lib/systemd/system/wmbusmeters.service; disabled; vendor >
     Active: active (running) since Mon 2024-04-01 15:47:45 CEST; 32min ago
       Docs: https://github.com/wmbusmeters/wmbusmeters
             man:wmbusmeters(1)
    Process: 376184 ExecStartPre=/bin/mkdir -p /var/lib/wmbusmeters/meter_readi>
    Process: 376185 ExecStartPre=/bin/chown -R wmbusmeters:wmbusmeters /var/lib>
    Process: 376186 ExecStartPre=/bin/mkdir -p /var/log/wmbusmeters (code=exite>
    Process: 376187 ExecStartPre=/bin/chown -R wmbusmeters:wmbusmeters /var/log>
    Process: 376188 ExecStartPre=/bin/mkdir -p /run/wmbusmeters (code=exited, s>
    Process: 376189 ExecStartPre=/bin/chown -R wmbusmeters:wmbusmeters /run/wmb>
    Process: 376190 ExecStart=/usr/sbin/wmbusmetersd /run/wmbusmeters/wmbusmete>
   Main PID: 376191 (wmbusmetersd)
      Tasks: 8 (limit: 779)
        CPU: 35min 6.911s
     CGroup: /system.slice/wmbusmeters.service
             ├─376191 /usr/sbin/wmbusmetersd /run/wmbusmeters/wmbusmeters.pid
             ├─376205 /bin/sh -c ERRFILE=$(mktemp -t wmbusmeters_rtlsdr.XXXXXXX>
             ├─376208 tail -f /tmp/wmbusmeters_rtlsdr.2FDDmgBVEhCh
             ├─376209 /usr/bin/rtl_sdr -d 0 -f 868.95M -s 1.6e6 -
             └─376210 /usr/bin/rtl_wmbus -f

Apr 01 15:47:45 raspberrypi systemd[1]: Starting "wmbusmeters service"...
Apr 01 15:47:45 raspberrypi wmbusmetersd[376190]: (wmbusmeters) started /run/wm>
Apr 01 15:47:45 raspberrypi systemd[1]: Started "wmbusmeters service".
```

and the log file shows me

```[2024-04-01_15:47:45] Started config rtlwmbus[00000001] listening on t1,c1 using fq 868.95M
(wmbusmeters) waiting for telegrams
No meters configured. Printing id:s of all telegrams heard!
[2024-04-01_15:47:47] (memory) rss 4653056 peak 4.43 MiB
```

Using rtl_test, as soon as started wmbusmeters, results in the following error:

```Found 1 device(s):
  0:  RTLSDRBlog, Blog V4, SN: 00000001

Using device 0: Generic RTL2832U OEM
usb_claim_interface error -6
Failed to open rtlsdr device #0.
```

I also went the frequency up and down by 0.05 MHz, as I saw in another post, but nothing happened.

Thanks in advance!

**Comment by Milvus11** [comment]
> Hey guys, I am a real newbie, I've been searching through the internet, also used AI for help, but don't receive any telegrams for my Techem FM radio 4 water meters, nor my vario 4 Typ 4.1.2 floor heating meter, though trying it for two weeks now. I had some trouble installing it correctly so I made myself a manual for installation via make, to verify it was right and nothing went wrong there already, I attach it here as well. Underneath you will find the status and log output. Can someone tell me if/what I did wrong?
> 
> ### 1. System update
> `sudo apt update` `sudo apt upgrade -y`
> 
> ### 2. Install dependencies
> `sudo apt install -y git cmake build-essential librtlsdr-dev libusb-1.0-0-dev`
> 
> ### 3. Install RTL-SDR
> Blocking intern DBT-T `echo "blacklist dvb_usb_rtl28xxu" | sudo tee /etc/modprobe.d/blacklist-rtl.conf ` Reboot Raspi `sudo reboot` Install rtl-sdr `sudo apt install -y rtl-sdr` Test, if RTL-SDR dongle is recognised `rtl_test`
> 


I got an Update to this: I added --verbose and checked output wit wmbusmeters --logtelegrams rtlwmbus.
For roughly 20 mins, I receive the output:

'''
Started config rtlwmbus[00000001] listening on none
No meters configured. Printing id:s of all telegrams heard!
Lost 00000001 closing rtlwmbus[00000001]
Started config rtlwmbus[00000001] listening on none
Lost 00000001 closing rtlwmbus[00000001]
Started config rtlwmbus[00000001] listening on none
Lost 00000001 closing rtlwmbus[00000001]
Started config rtlwmbus[00000001] listening on none
Lost 00000001 closing rtlwmbus[00000001]
Started config rtlwmbus[00000001] listening on none
Lost 00000001 closing rtlwmbus[00000001]
Started config rtlwmbus[00000001] listening on none
'''

repeatedly, then suddenly I receive:

`[ALARM SpecifiedDeviceNotFound] the device rtlwmbus is not working`

wmbusmeters status says, it is still active. When I leave wmbusmeters --logtelegrams rtlwmbus and start it again, it starts from beginning, and 20 mins later, it again shows, that the device is not working.

I read in another issue, that rtl_sdr could be involved in that? 
I am shortly before reinstalling the whole raspi once more, trying it with Snap and then just let it be, I don't get it.

**Comment by heziegl** [comment]
had similar problem (everything looked good, but no telegrams at all).
my solution was to install the v4 specific rtl-sdr package, see https://www.rtl-sdr.com/v4/
